From 44efa89a5b3e3f2506266c5a0d604874b2a8ff3e Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 26 May 2023 10:06:56 +0200
Subject: [PATCH 0531/1695] plat-stm32mp1: map DDR bottom page for STM32MP13 as
 well

The first page of the DDR, where the training area backup data will be
stored must be mapped as non-secure memory

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git a777d285e5ad31299c27179fee3d8cbc04bc54f2 plat-stm32mp1: map DDR bottom page for STM32MP13 as well]
Change-Id: I1adb4d2d54aeb761938f414ea6ab78ff3199dbc1
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/358377
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Tested-by: Pascal PAILLET <p.paillet@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp1/main.c b/core/arch/arm/plat-stm32mp1/main.c
index 92d6a1aed..05394f844 100644
--- a/core/arch/arm/plat-stm32mp1/main.c
+++ b/core/arch/arm/plat-stm32mp1/main.c
@@ -50,10 +50,10 @@ register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE, GIC_SIZE);
 
 register_ddr(DDR_BASE, CFG_DRAM_SIZE);
 
-#ifdef CFG_STM32MP15
 /* Map non-secure DDR bottom for the low power sequence */
 register_phys_mem(MEM_AREA_RAM_NSEC, DDR_BASE, SMALL_PAGE_SIZE);
 
+#ifdef CFG_STM32MP15
 /* Map TEE physical RAM as read-only for content storage when suspending */
 register_phys_mem(MEM_AREA_ROM_SEC, TEE_RAM_START, TEE_RAM_PH_SIZE);
 #endif /* CFG_STM32MP15 */
-- 
2.47.2

